packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,232 @@
+            if (this.findCommand(command, cwd) === undefined) {
+                throw (`Command not found: ${command}`);
+            }
+
+            // use terminal or raw process
+            if (options.processType === 'terminal') {
+                this.logger.info('Task: creating underlying terminal process');
+                proc = this.terminalProcessFactory(<TerminalProcessOptions>processOptions);
+            } else {
+                this.logger.info('Task: creating underlying raw process');
+                proc = this.rawProcessFactory(<RawProcessOptions>processOptions);
+            }
+
+            task = new Task(
+                this.taskManager,
+                proc,
+                options.errorMatcherName
+            );
+
+            // when underlying process exits, notify tasks listeners
+            const toDispose = new DisposableCollection();
+
+            toDispose.push(
+                proc.onExit(event => {
+                    this.fireTaskExitedEvent({
+                        'taskId': task.id,
+                        'code': event.code,
+                        'signal': event.signal
+                    });
+                })
+            );
+
+            this.taskToDispose.set(task.id, toDispose);
+
+            return Promise.resolve(
+                {
+                    taskId: task.id,
+                    osProcessId: proc.pid,
+                    terminalId: (options.processType === 'terminal') ? proc.id : undefined,
+                    processId: proc.id
+                }
+            );
+
+        } catch (error) {
+            this.logger.error(`Error occured while creating task: ${error}`);
+            return Promise.reject(new Error(error));
+        }
+    }
+
+    protected fireTaskExitedEvent(event: ITaskExitedEvent) {
+        this.logger.debug(log => log(`task has exited:`, event));
+        if (this.client) {
+            this.client.onTaskExit(event);
+        }
+    }
+
+    public tasksListChanged(tasks: string[]) {
+        this.fireTaskChangedEvent({ 'tasks': tasks });
+    }
+
+    protected fireTaskChangedEvent(event: ITasksChangedEvent) {
+        this.logger.debug(log => log(`Configured tasks changed:`, event));
+        if (this.client) {
+            this.client.onTasksConfigChanged(event);
+        }
+    }
+
+    kill(id: number): Promise<void> {
+        const taskToKill = this.taskManager.get(id);
+        if (taskToKill !== undefined) {
+            taskToKill.kill();
+        }
+        return Promise.resolve();
+    }
+
+    /** Set the client we send the notifications-to */
+    setClient(client: ITaskClient | undefined) {
+        this.client = client;
+    }
+
+    // Check if task command can be found, if so return its resolved
+    // absolute path, else returns undefined
+    private findCommand(command: string, cwd: string): string | undefined {
+        const systemPath = process.env.PATH;
```

Comment:
not sure this is windows friendly?

Reply:
I just mean I'm not sure PATH is the env var windows uses.

So you mean if need to test cmd.exe and the destination ? I think we can assume cmd.exe exists no ?

Reply:
Maybe we can assume that cmd.exe exists. The user could in theory point towards an alternate interpreter. The destination can be a script/executable but can also be an interpreter built-in, like "DIR". So we may not find it as a file in that case, even if we look.

Reply:
indeed that's just the situation I wanted to avoid.

What if we were to try to run it, and after that if it fails try to narrow down the reason using findCommand?

That way these cases would be avoided most likely.

Reply:
Interesting idea; I like it. This way we at least try to execute the command, even if we can't figure-out exactly what it's calling. If it succeeds, everything is good. If it fails, we dig down to try and see if we can figure-out why it failed, looking into locating the command, maybe in some cases its file parameter(s). 

I do not think we avoid the corner cases entirely, however. There will still be cases like : "cmd.exe /c DIZ" that will be hard to pin down: did the user want the "DIR" built-in of cmd.exe and made a typo?  Or was he trying to execute a DIZ.bat batch file in cwd?